Wrappers/Python/cil/optimisation/functions/L1Sparsity.py

```diff
@@ -0,0 +1,152 @@
+        self.Q = Q
+
+        self.l1norm = L1Norm(b=b, weight=weight)
+
+    def __call__(self, x):
+        r"""Returns the value of the L1Sparsity function at x.
+
+        Consider the following cases:
+
+        a) .. math:: F(x) = ||Qx||_{1}
+        b) .. math:: F(x) = ||Qx - b||_{1}
+
+        In the weighted case, `weight` = :math:`w` is an array of positive weights.
+
+        a) .. math:: F(x) = ||Qx||_{L^1(w)}
+        b) .. math:: F(x) = ||Qx - b||_{L^1(w)}
+
+        with :math:`||x||_{L^1(w)} = || x w||_1 = \sum_{i=1}^{n} |x_i| w_i`.
+            
+        """
+        y = self.Q.direct(x)
+        return self.l1norm(y)
+
+    def convex_conjugate(self, x):
+        r"""Returns the value of the convex conjugate of the L1Sparsity function at x.
+        Here, we need to use the convex conjugate of L1Sparsity, which is the Indicator of the unit 
+        :math:`\ell^{\infty}` norm on the operator domain. (Since Q is a basis of L^2).
+
+
+        Consider the non-weighted case: 
+
+
+        a) .. math:: F^{*}(x^{*}) = \mathbb{I}_{\{\|\cdot\|_{\infty}\leq1\}}(Qx^{*})
+        b) .. math:: F^{*}(x^{*}) = \mathbb{I}_{\{\|\cdot\|_{\infty}\leq1\}}(Qx^{*}) + \langle Qx^{*},b\rangle
+
+
+        .. math:: \mathbb{I}_{\{\|\cdot\|_{\infty}\leq1\}}(x^{*})
+            = \begin{cases}
+            0, \mbox{if } \|x^{*}\|_{\infty}\leq1\\
+            \infty, \mbox{otherwise}
+            \end{cases}
+
+        In the weighted case the convex conjugate is the indicator of the unit
+        :math:`L^{\infty}` norm.
+
+        See:
+        https://math.stackexchange.com/questions/1533217/convex-conjugate-of-l1-norm-function-with-weight
+
+        a) .. math:: F^{*}(x^{*}) = \mathbb{I}_{\{\|\cdot\|_{L^\infty(w^{-1})}\leq 1\}}(Qx^{*})
+        b) .. math:: F^{*}(x^{*}) = \mathbb{I}_{\{\|\cdot\|_{L^\infty(w^{-1})}\leq 1\}}(Qx^{*}) + \langle Qx^{*},b\rangle
+
+        with :math:`\|x\|_{L^\infty(w^{-1})} = \max_{i} \frac{|x_i|}{w_i}`.
+    
+    
+        """
+        y = self.Q.direct(x)
+        return self.l1norm.convex_conjugate(y)
+
+    def proximal(self, x, tau, out=None):
+
+        r"""Returns the value of the proximal operator of the L1 Norm function at x with scaling parameter `tau`.
+
+
+        Consider the following cases:
+
+        a) .. math:: \mathrm{prox}_{\tau F}(x) = Q^T \mathrm{ShinkOperator}_{\tau}(Qx)
+        b) .. math:: \mathrm{prox}_{\tau F}(x) = Q^T \mathrm{ShinkOperator}_\tau(Qx) + b
+
+        where,
+
+        .. math :: \mathrm{prox}_{\tau | \cdot |}(x) = \mathrm{ShinkOperator}(x) = sgn(x) * \max\{ |x| - \tau, 0 \}
+
+        The weighted case follows from Example 6.23 in Chapter 6 of "First-Order Methods in Optimization"
+        by Amir Beck, SIAM 2017 https://archive.siam.org/books/mo25/mo25_ch6.pdf
+
+        a) .. math:: \mathrm{prox}_{\tau F}(x) = Q^T \mathrm{ShinkOperator}_{\tau*w}(Qx)
+        b) .. math:: \mathrm{prox}_{\tau F}(x) = Q^T \mathrm{ShinkOperator}_{\tau*w}(Qx) + b
```

Comment:
@tommheik  does this need to be:
```suggestion
        b) .. math:: \mathrm{prox}_{\tau F}(x) = Q^T \left( \mathrm{ShinkOperator}_{\tau*w}(Qx) + b \right)
```

Reply:
Same as above.